dlp/snippets/src/main/java/dlp/snippets/DeIdentifyDataReplaceWithDictionary.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+// [START dlp_deidentify_dictionary_replacement]
+
+
+import com.google.cloud.dlp.v2.DlpServiceClient;
+import com.google.privacy.dlp.v2.ContentItem;
+import com.google.privacy.dlp.v2.CustomInfoType.Dictionary.WordList;
+import com.google.privacy.dlp.v2.DeidentifyConfig;
+import com.google.privacy.dlp.v2.DeidentifyContentRequest;
+import com.google.privacy.dlp.v2.DeidentifyContentResponse;
+import com.google.privacy.dlp.v2.InfoType;
+import com.google.privacy.dlp.v2.InfoTypeTransformations;
+import com.google.privacy.dlp.v2.InspectConfig;
+import com.google.privacy.dlp.v2.LocationName;
+import com.google.privacy.dlp.v2.PrimitiveTransformation;
+import com.google.privacy.dlp.v2.ReplaceDictionaryConfig;
+import java.io.IOException;
+
+public class DeIdentifyDataReplaceWithDictionary {
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // The string to de-identify
+    String textToDeIdentify =
+        "My name is Alicia Abernathy, and my email address is aabernathy@example.com.";
+    deidentifyDataReplaceWithDictionary(projectId, textToDeIdentify);
+  }
+
+  public static void deidentifyDataReplaceWithDictionary(String projectId, String textToDeIdentify)
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DlpServiceClient dlp = DlpServiceClient.create()) {
+      // Specify the content to be inspected.
+      ContentItem item = ContentItem.newBuilder().setValue(textToDeIdentify).build();
+
+      // Specify the type of info the inspection will look for.
+      // See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+      InfoType infoType = InfoType.newBuilder().setName("EMAIL_ADDRESS").build();
+      InspectConfig inspectConfig = InspectConfig.newBuilder().addInfoTypes(infoType).build();
+
+      WordList wordList =
+          WordList.newBuilder().addWords("izumi@example.com").addWords("alex@example.com").build();
```

Comment:
Add a comment explaining what these values are.

Specifically, it would be good to call out that `EMAIL_ADDRESS`es found by inspection will be replaced with a randomly selected value from this list.